tests/dps/test_single_process_restart.py

```diff
@@ -0,0 +1,169 @@
+from conftest import LOG_DIR
+from libs.dps.dtm0_lib import DTM0TestLib
+
+
+class TestSingleProcessRestart:
+    """Test Class for Single Process Restart."""
+
+    @classmethod
+    def setup_class(cls):
+        """Setup class."""
+
+        cls.log = logging.getLogger(__name__)
+        cls.num_nodes = len(CMN_CFG["nodes"])
+        cls.master_node_list = []
+        cls.worker_node_list = []
+        for node in CMN_CFG['nodes']:
+            node_obj = LogicalNode(hostname=node["hostname"],
+                                   username=node["username"],
+                                   password=node["password"])
+            if node["node_type"].lower() == "master":
+                cls.master_node_list.append(node_obj)
+            else:
+                cls.worker_node_list.append(node_obj)
+        cls.setup_type = CMN_CFG["setup_type"]
+        cls.test_completed = False
+        cls.health_obj = Health(cls.master_node_list[0].hostname,
+                                cls.master_node_list[0].username,
+                                cls.master_node_list[0].password)
+        cls.test_cfg = configmanager.get_config_wrapper(fpath="config/test_dtm0_config.yaml")
+        cls.m0d_process = 'm0d'
+        cls.restart_wait_time_secs = 10
+        cls.dtm0_obj = DTM0TestLib()
+
+    def teardown_method(self):
+        """Teardown class method."""
+        if not self.test_completed:
+            self.log.info("Test Failure observed, collecting support bundle")
+            path = os.path.join(LOG_DIR, LATEST_LOG_FOLDER)
+            resp = support_bundle_utils.collect_support_bundle_k8s(local_dir_path=path,
+                                                                   scripts_path=K8S_SCRIPTS_PATH)
+            assert_utils.assert_true(resp)
+        # TODO : Redeploy setup after test completion.
+
+    @pytest.mark.lc
+    @pytest.mark.dtm0
+    @pytest.mark.tags("TEST-41204")
+    def test_read_during_m0d_restart(self):
+        """Verify READ during m0d restart using pkill."""
+        self.log.info("STARTED: Verify READ during m0d restart using pkill")
+        bucket_name = 'bucket-test-41204'
+        object_name = 'object-test-41204'
+        test_41204_cfg = self.test_cfg['test_41204']
+        que = multiprocessing.Queue()
+
+        self.log.info("Step 1: Perform write Operations :")
+        self.dtm0_obj.perform_write_op(bucket_prefix=bucket_name,
+                                       object_prefix=object_name,
+                                       no_of_clients=test_41204_cfg['clients'],
+                                       no_of_samples=test_41204_cfg['samples'],
+                                       obj_size=test_41204_cfg['size'],
+                                       log_file_prefix=test_41204_cfg['log_file_prefix'], queue=que)
+        resp = que.get()
+        assert_utils.assert_true(resp[0], resp[1])
+        workload_info = resp[1]
+
+        self.log.info("Step 2: Perform Read Operations :")
```

Comment:
so this is read ops on previously written data right? so can we write like:
Start Read Operations on previously written data in background

Reply:
Done